### PR TITLE
App scaffold

### PR DESCRIPTION
**Pull request by salvanicola**
- Define solution structure (Clean architecture)
- Entities
- Sketched scaffolding
- App architecture in [wiki](https://github.com/Kuama-IT/kerbero/wiki/Project-design) proposal

### CODE REVIEW

**Comment by kuamanet** [comment]
@salvanicola please drop the domain folder

**Review by kuamanet — CHANGES_REQUESTED** [review]
Please drop the domain folder

**Comment by salvanicola** [comment]
> 

All the folder or just the content?

File: Kerbero.NET/src/WebApi/Program.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+builder.Services.AddApplicationServices();
+builder.Services.AddInfrastructureServices(builder.Configuration);
+builder.Services.AddWebUIServices();
```
Comment (vash72) [inline Kerbero.NET/src/WebApi/Program.cs:6]: WebUI? but the project is WebApi (HTTP JSON Restfull API) so why UI?

File: Kerbero.NET/src/WebApi/appsettings.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "UseInMemoryDatabase": true,
+  "ConnectionStrings": {
+    "Server": "127.0.0.1,1433",
+    "DbName": "kerbero_db",
+    "User": "sa",
+    "Password": "Kuama123"
```
Comment (vash72) [inline Kerbero.NET/src/WebApi/appsettings.json:7]:
```suggestion

```
sensible data should never be tracked under source control

File: Kerbero.NET/src/WebApi/Program.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+// Configure the HTTP request pipeline.
+if (app.Environment.IsDevelopment())
+{
+    app.UseSwagger();
+    app.UseSwaggerUI();
+}
+
+app.UseHttpsRedirection();
+
+app.UseAuthorization();
```
Comment (vash72) [inline Kerbero.NET/src/WebApi/Program.cs:25]: why?

**Review by vash72 — CHANGES_REQUESTED** [review]

File: Kerbero.NET/src/Infrastructure/ConfigureServices.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+public static class ConfigureServices
+{
+    public static IServiceCollection AddInfrastructureServices(this IServiceCollection services, IConfiguration configuration)
```
Comment (vash72) [inline Kerbero.NET/src/Infrastructure/ConfigureServices.cs:7]:
```suggestion
    public static IServiceCollection AddInfrastructureServices(this IServiceCollection)
```
you should retrieve IConfiguration form DI where you need, not passing it as a parameter here

**Review by vash72 — CHANGES_REQUESTED** [review]

File: src/Kerbero.WebApi/Controllers/WeatherForecastController.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
+using Microsoft.AspNetCore.Mvc;
```
Comment (kuamanet) [inline src/Kerbero.WebApi/Controllers/WeatherForecastController.cs:1]: We don't need this controller, right?
Either flag this with a 
```csharp
// TODO remove this
```
or directly remove this file

File: src/Kerbero.WebApi/Program.cs [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+// Configure the HTTP request pipeline.
+if (app.Environment.IsDevelopment())
+{
+    app.UseSwagger();
+    app.UseSwaggerUI();
+}
+
+app.UseHttpsRedirection();
+
+app.UseAuthorization();
```
Comment (kuamanet) [inline src/Kerbero.WebApi/Program.cs:21]: No need fot authorization right now

File: tests/Kerbero.WebApiTests/UnitTest1.cs [hunk 7]
```diff
@@ -0,0 +1,10 @@
+namespace Kerbero.WebApiTests;
+
+public class UnitTest1
+{
+    [Fact]
+    public void Test1()
+    {
```
Comment (kuamanet) [inline tests/Kerbero.WebApiTests/UnitTest1.cs:7]:
```suggestion
    {
        Assert.True(true);
```

File: src/Kerbero.WebApi/WeatherForecast.cs [hunk 8]
```diff
@@ -0,0 +1,12 @@
+namespace Kerbero.WebApi;
```
Comment (kuamanet) [inline src/Kerbero.WebApi/WeatherForecast.cs:1]: We don't need a `WeatherForecast`, right?

**Review by kuamanet — CHANGES_REQUESTED** [review]

File: .github/workflows/dotnet.yml [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: ${{ matrix.dotnet-version }}
+      - name: Restore dependencies
+        run: dotnet restore
+      - name: Build
+        run: dotnet build --no-restore
```
Comment (vash72) [inline .github/workflows/dotnet.yml:26]:
```suggestion
      - name: Build
        run: dotnet build
```
=)

File: .github/workflows/dotnet.yml [hunk 10]
```diff
@@ -0,0 +1,28 @@
+name: .NET
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
```
Comment (vash72) [inline .github/workflows/dotnet.yml:7]: this workflow seems to only for pullrequests, why running it on master push?

File: .github/workflows/dotnet.yml [hunk 11]
```diff
@@ -0,0 +1,28 @@
+name: .NET
```
Comment (vash72) [inline .github/workflows/dotnet.yml:1]: 😕 , the name of the file is not great, it should something like "pr-integration"

**Review by vash72 — CHANGES_REQUESTED** [review]

File: .github/workflows/dotnet.yml (hunk 9 above)
Comment (salvanicola) [reply .github/workflows/dotnet.yml:26]: You trust dotnet too much

File: README.md [hunk 12]
```diff
@@ -1 +1,10 @@
 # kerbero
```
Comment (vash72) [inline README.md:1]: usually we write the how to start the project in the readme

**Review by vash72 — DISMISSED** [review]
not rly sure about the test project name

File: .github/workflows/pr-integration.yml [hunk 13]
```diff
@@ -0,0 +1,24 @@
+name: pr-integration
```
Comment (kuamanet) [inline .github/workflows/pr-integration.yml:1]:
```suggestion
name: Build and Test
```

File: README.md (hunk 12 above)
Comment (kuamanet) [reply README.md:1]: Yup, @salvanicola add a section like
```markdown
# How to run the project
....
```

**Review by kuamanet — CHANGES_REQUESTED** [review]
Couple of things and we're done

**Review by kuamanet — APPROVED** [review]

**Pull request merged**